src/components/ViewRequest/Actions/CancelRequest.tsx

```diff
@@ -10,21 +10,32 @@ import {
   Tooltip,
 } from "@fluentui/react-components";
 import { DeleteIcon } from "@fluentui/react-icons-mdl2";
-import { useDeleteRequest } from "api/requestsApi";
+import { useCancelRequest, useRequest } from "api/requestsApi";
 import { Navigate, useParams } from "react-router-dom";
 
-const DeleteRequest = () => {
+declare const _spPageContextInfo: {
+  userId: number;
+};
+
+const CancelRequest = () => {
   const params = useParams();
-  const deleteRequest = useDeleteRequest();
+  const request = useRequest(Number(params.requestId));
+  const cancelRequest = useCancelRequest();
+
+  const blockedStages = ["Complete", "Cancelled"];
+
+  const canCancel =
+    Number(request.data?.Author?.Id) === _spPageContextInfo.userId &&
+    !blockedStages.includes(request.data?.stage || "");
 
-  const deleteHandler = () => {
-    deleteRequest.mutate(Number(params.requestId));
+  const cancelHandler = () => {
+    cancelRequest.mutate(Number(params.requestId));
   };
 
   return (
     <>
-      {deleteRequest.isSuccess && <Navigate to="/" />}
-      {!deleteRequest.isSuccess && (
+      {cancelRequest.isSuccess && <Navigate to="/" />}
+      {!cancelRequest.isSuccess && (
         <Dialog modalType="alert">
           <DialogTrigger disableButtonEnhancement>
             <Tooltip withArrow content="Delete" relationship="label">
```

Comment:
Text should be updated to "Cancel Request"
```suggestion
            <Tooltip withArrow content="Cancel Request" relationship="label">
```